tests/test_distributedarray_nccl.py

```diff
@@ -0,0 +1,438 @@
+"""Test the DistributedArray class
+Designed to run with n processes
+$ mpiexec -n 10 pytest test_distributedarray.py --with-mpi
+
+This file employs the same test sets as test_distributedarray under NCCL environment
+"""
+
+import numpy as np
+from mpi4py import MPI
+import pytest
+from numpy.testing import assert_allclose
+
+from pylops_mpi import DistributedArray, Partition
+from pylops_mpi.DistributedArray import local_split
+
+import cupy as cp
```

Comment:
This is fine to import this way as we know we will never run this test without `cupy`... in future we may however want to do something like this https://github.com/PyLops/pylops/pull/660 and we will eventually create a Github-action for the NCCL backend with the help of @hongyx11  like we are about to do for PyLops

But let's not do this until the NCCL API is a bit more evolved...